src/components/Calculator.js

```diff
@@ -1,58 +1,72 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import './Calculator.css';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.addNumber = this.addNumber.bind(this);
+    this.addOperation = this.addOperation.bind(this);
+  }
+
+  addNumber(event) {
+    const { next } = calculate(this.state, event.target.textContent);
+    this.setState({ next });
+  }
+
+  addOperation(event) {
+    const { total, next, operation } = calculate(this.state, event.target.textContent);
+    this.setState({ total, next, operation });
   }
```

Comment:
- [ ] After completing one operation Ex: `1+2 = 3`, I tried to do another operation ` + 9`, but it shows me the error of `TypeError: one.plus is not a function`. Please fix it for a smoother calculation.

The smoother calculation is:
```
1 + 2 = 3 then + 9 = 12
```